ckanext/stadtzhharvest/harvester.py

```diff
@@ -379,6 +406,31 @@ def _get_existing_package(self, package_dict):
             log.debug('Could not find pkg %s' % package_dict['name'])
         return existing_package
 
+    def _get_existing_packages_names(self, harvest_job):
+        context = self._create_new_context()
+        try:
+            existing_packages = get_action('package_search')(
+                context,
+                {'fq': 'harvest_source_id:"{0}"'.format(harvest_job.source_id)}
+            )
+            if (existing_packages['count']) > len(existing_packages):
+                existing_packages = get_action('package_search')(
+                    context,
+                    {'fq': 'harvest_source_id:"{0}"'.format(
+                        harvest_job.source_id
+                     ),
+                     'rows': existing_packages['count']}
+                )
```

Comment:
This approach will fail if more than 1000 datasets are found, [since `1000` is the maximum rows that `package_search`](https://docs.ckan.org/en/ckan-2.7.3/api/#ckan.logic.action.get.package_search) can return. Instead of simply making 2 requests for the same data, I would follow the pagination with something like that:

```python
n = 500
page = 1
existing_packages_names = []

while True:
    search_params = {
        'fq': 'harvest_source_id:"{0}"'.format(harvest_job.source_id )
        'rows': n,
        'start': n * (page - 1),
    }
    existing_packages  = get_action('package_search')(context, search_params)
    if len(existing_packages):
        existing_packages_names.extend([pkg['name'] for pkg in existing_packages['results']])
        page = page + 1
    else:
        break
```

Like that, you get 500 records in each request, and use around 1-3 iterations to get all packages.
I know that currently all our harvesters have less than 1000 records, but I think this solution is future-proof, since we expect more datasets, especially from the GEOPORTAL harvesters in the near future.

Reply:
Done.

Reply:
Any particular reason to use `n = 2`? I'd use a higher value like 500 to reduce the amount of calls to package_search

Reply:
Sorry that was for debugging purposes. I just forgot to change it back. Will do that immediately. Well spotted, thank you very much.